tests/tests_test.go

```diff
@@ -0,0 +1,218 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path"
+	"strings"
+	"time"
+
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"github.com/konveyor/mig-controller/pkg/pods"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ctx = context.Background()
+
+var _ = Describe("Running migmigration when storage, cluster and plan are ready", func() {
+	var (
+		migPlan      *migapi.MigPlan
+		migPlanName  string
+		namespaces   []string
+		testFilePath string
+	)
+
+	JustBeforeEach(func() {
+		log.Println("creating migplan")
+		Expect(hostClient.Create(ctx, migPlan)).Should(Succeed())
+		Eventually(func() bool {
+			log.Println("waiting for migPlan to be ready")
+			Expect(hostClient.Get(ctx, client.ObjectKey{Name: migPlanName, Namespace: MigrationNamespace}, migPlan)).Should(Succeed())
+			return migPlan.Status.IsReady()
+		}, time.Minute*5, time.Second).Should(Equal(true))
+	})
+
+	AfterEach(func() {
+		err = hostClient.Delete(ctx, &migapi.MigPlan{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      migPlanName,
+				Namespace: MigrationNamespace,
+			},
+		})
+		if err != nil {
+			log.Println(err)
+		}
+	})
+
+	Context("Testing BZ #1965421, PVC name is more than 63 char long", func() {
+
+		BeforeEach(func() {
+			migPlanName = "e2e-1965421"
+			namespaces = []string{"ocp-41583-longpvcname"}
+			testFilePath = "data/test/"
+			migPlan = NewMigPlan(namespaces, migPlanName)
+
+			_, err = sourceClient.CoreV1().Namespaces().Create(ctx, NewMigrationNS("ocp-41583-longpvcname"), metav1.CreateOptions{})
+			if errors.IsAlreadyExists(err) {
+				log.Println(err)
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			_, err = sourceClient.CoreV1().PersistentVolumeClaims("ocp-41583-longpvcname").Create(ctx, NewPVC("long-name-123456789011121314151617181920212223242526272829303132", "ocp-41583-longpvcname"), metav1.CreateOptions{})
+
+			if errors.IsAlreadyExists(err) {
+				log.Println(err)
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			_, err = sourceClient.AppsV1().Deployments("ocp-41583-longpvcname").Create(ctx, NewDeploymentFor41583(), metav1.CreateOptions{})
+			if errors.IsAlreadyExists(err) {
+				log.Println(err)
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			Eventually(func() v1.PodPhase {
+				podList, err := sourceClient.CoreV1().Pods("ocp-41583-longpvcname").List(ctx, metav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				for _, p := range podList.Items {
+					if p.Status.Phase == "Running" {
+						podCmd := pods.PodCommand{
+							Pod:     &p,
+							RestCfg: sourceCfg,
+							Args:    []string{"sh", "-c", "dd if=/dev/zero of=/" + testFilePath + "file_1 bs=60M count=1"},
```

Comment:
@JaydipGabani I recall from testing pvc-migrate that certain classes of generated data lead to a no-op during rsync transfer. For this test to be valid, it may be worth generating random data (rather than a file of 0s), checksumming that, then verifying the checksum matches on the destination.

Take a look at this DataGenerator for an example of generating data that _won't_ result in an Rsync no-op: https://github.com/konveyor/pvc-migrate/tree/master/benchmarks/DataGenerator

@pranavgaikwad does this match your memory?

Reply:
@djwhatle yeah, zero data is not reliable test. Can replace with urandom() instead. Also, I suggested to use a pre-populated volume in previous comment
> Is it possible to populate a volume using a temporary pod as a preparation step of the test, and then attaching that same volume in a dummy Pod? so that the Pod being migrated doesn't have to create its own data. We can verify whether we find the same data in the migrated Pod that we created in the prepare step using checksum.

Reply:
@djwhatle @pranavgaikwad I have added verification for checksum for migrated files and now using dataGenerator instead of `dd` command.